src/ripple/rpc/impl/TxMetaSerializer.cpp

```diff
@@ -0,0 +1,57 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2019 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/rpc/TxMetaSerializer.h>
+
+#include <ripple/app/ledger/LedgerMaster.h>
+#include <ripple/app/ledger/OpenLedger.h>
+#include <ripple/app/misc/Transaction.h>
+#include <ripple/ledger/View.h>
+#include <ripple/net/RPCErr.h>
+#include <ripple/protocol/AccountID.h>
+#include <ripple/protocol/Feature.h>
+#include <ripple/rpc/Context.h>
+#include <ripple/rpc/DeliveredAmount.h>
+#include <ripple/rpc/NFTokenID.h>
+#include <ripple/rpc/NFTokenOfferID.h>
+#include <ripple/rpc/TxMetaSerializer.h>
+#include <ripple/rpc/impl/RPCHelpers.h>
+#include <boost/algorithm/string/case_conv.hpp>
+
+namespace ripple {
+namespace RPC {
+
+void
+serializeTxMetaAsJSON(
```

Comment:
This function is supposed to be a general purpose function, as it inserts a `delivered_amount` field which is not NFT related. 

This function would be the place to add synthetic fields, but at the moment there are not many of them and 2 of them happen to be NFT-related

Reply:
@scottschurr yeah i agree delivered amount is used in other places and it can be taken out of this function. But wouldn’t this function be better suited for future non-NFT synthetic fields if it was named in a general purpose manner ? (I don’t know what new synthetic fields might be added, just just saying it’s possible)

Reply:
Right, this _might_ be a good place to handle future synthetic fields.  But we don't know yet.  We don't know the (future) situation.  If that work happens in an area that does not include NFTs, it's unlikely that the developer will even be aware this function exists.  So they probably won't even consider expanding this function.  But, if they do consider doing that, it would make sense for them to change the name at that time.  This function is not going into a library.  We have access to the source code for all of the callers.  If someone wants to change the name in the future they can do so easily.

For now, I think it's better if the name reflects what the function does right now.  It makes less sense to me to name it for future aspirations.